### PR TITLE
Cursor scaling relativity fix

### PR DESCRIPTION
**Pull request by mpkoz**
## Overview

The current implementation of cursor scaling based on distance is achieved by an angular scale calculation. The calculation (2 * distance * Tan(angle * 0.5f)) assumes the object has a diameter of 1 unit. In order to make this calculation relevant to the child meshes under the DefaultCursor prefab, I initially (in Commit 7ea09f90c718208e832bd9c41cd79f3c0d739dc6) calculate the bounds of the active child mesh on Start() and divide the results by 2 * magnitude of the bounding box of the mesh. In other words, the approximate diagonal diameter of the mesh. This works as expected and gives a relativity to the original angular scale calculation. At first, this seemed like the right way to go. 

However, due to an error I made in calculating the bounding box diagonal magnitude, adding additional cursors to the scene after scene initialization when the user has moved from the world origin, the scaling of the new cursors is not correct. The fix for this was to calculate the bounding box diagonal on the mesh itself via mesh.bounds (local space) rather than the current method of renderer.bounds (world space). The issue becomes more pronounced the further they are from the origin. 

Instead of pushing that fix, I believe this issue and future issues can be avoided by using the angular scale function without a denominator. This would make things easier for a third party developer to grasp as well. Any child meshes underneath the DefaultCursor prefab, added dynamically or by default, will be assumed to have a bounds of (1,1,1). The calculation of an initial bounding box was unnecessary to begin with for the amount of difficulty it might cause in the future.

This PR is a pretty straight forward. It removes the denominator and no longer calculates any bounding boxes. Any child meshes under the DefaultCursor prefab will be scaled according to their localScale in the hierarchy, as opposed to their model-space bounding box scale. The only effective difference is a default value of 50.0 instead of 0.65 in the public CursorAngularScale variable. All appearances are the same as 7ea09f90c718208e832bd9c41cd79f3c0d739dc6, hence no screenshots.

## Changes
- Fixes: # https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6365

### CODE REVIEW

**Comment by keveleigh** [comment]
>adding additional cursors to the scene after scene initialization when the user has moved from the world origin, the scaling of the new cursors is not correct.

It'd be great to add a test for this case too!

**Comment by mpkoz** [comment]
Is a test necessary? Happy to, but since the omission of the bounding box calculation, none of the functionality has anything to do with world vs local positions. Therefore not sure what a test would accomplish.

**Comment by julenka** [comment]
> Is a test necessary? Happy to, but since the omission of the bounding box calculation, none of the functionality has anything to do with world vs local positions. Therefore not sure what a test would accomplish.

I think it would be great to have a test that verifies "when the cursor is two meters away, its size is X", when a cursor is 0.5 meters away, its size is Y". The reason is actually not as much to verify that your code works now, but to prevent this behavior from breaking in the future. Let's say somebody makes a new feature that breaks this behavior. It would be great if that person can't check in until the test that verifies cursors scaling correctly passes. Please see one of principles "[prioritize writing tests](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#prioritize-writing-tests)" for more details.

**Comment by mpkoz** [comment]
Ok! Will do.

**Comment by mpkoz** [comment]
Ok, wrote a more comprehensive test than the one checked in previously. Also changed some of the variable names to better match common mathematical terminology.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs [hunk 1]
```diff
@@ -113,17 +113,17 @@ public bool ResizeCursorWithDistance
         private bool resizeCursorWithDistance = false;
 
         /// <summary>
-        /// The value, in angle, of expected cursor size in relation to Main Camera
+        /// The angular scale of cursor in relation to Main Camera, assuming a mesh with bounds of Vector3(1,1,1)
         /// </summary>
-        public float CursorAngularScale
+        public float CursorAngularSize
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: If possible, please re-add a `public float CursorAngularScale` with the `Obsolete` tag (can just call directly into this property), so we don't break people currently calling this public property.

File: Assets/MixedRealityToolkit/Utilities/MathUtilities.cs [hunk 2]
```diff
@@ -49,10 +49,10 @@ public static Vector3 TransformVectorFromTo(Transform from, Transform to, Vector
         /// Retrieve angular measurement describing how large a sphere or circle appears from a given point of view.
         /// Takes an angle (at given point of view) and a distance and returns the actual diameter of the object.
         /// </summary>
-        public static float AngularScaleFromDistance(float angle, float distance)
+        public static float ScaleFromAngularSizeAndDistance(float angle, float distance)
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/MathUtilities.cs]: If possible, please re-add a `public static float ScaleFromAngularSizeAndDistance` with the `Obsolete` tag (can just call directly into this method), so we don't break people currently calling this public method.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs (hunk 1 above)
Comment (mpkoz) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: By "can just call directly into this property" do you mean [Obsolete]CursorAngularScale will get and set cursorAngularSize?

Comment (keveleigh) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: Yep! `cursorAngularSize` or `CursorAngularSize`, no need to add a separate backing field

Comment (mpkoz) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: Thanks Kurtis, I just pushed an update. Let me know if what I did checks out.

Comment (keveleigh) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: Looks great, thanks!

**Pull request merged**